libbeat/common/mapstr.go

```diff
@@ -138,6 +138,15 @@ func (m MapStr) Put(key string, value interface{}) (interface{}, error) {
 	return walkMap(key, m, mapStrOperation{putOperation{value}, true})
 }
 
+// SetValue dereferences a value if it is a pointer. Sets the dereferenced value if not nil.
```

Comment:
> What happens in the case that Put("a.b.c", nil) is used

```
func TestEmptyMap(t *testing.T) {
	m := MapStr{}
	m.Put("a.b.c", nil)
	t.Log(ConvertToGenericEvent(m).StringToPrint())
}
```

```
{
  "a": {
    "b": {}
  }
}
```

So we need some additional checks in the normalization code to make the whole tree disappear.